datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -114,23 +127,37 @@ def get_where_filter(where_filter_input, entity_name):
                 conditional_where_filters[:-1],
                 conditional_where_filters[-1],
                 boolean_operator,
+                SearchAPIQuery(entity_name),
             )
             where_filters.append(nested)
         elif list(where_filter_input.keys())[0] == "text":
-            # TODO - we might want to move this to the data
-            # definitions at a later point
-            text_operator_fields = {
-                "datasets": ["title"],
-                "documents": ["title", "summary"],
-                "files": ["name"],
-                "instrument": ["name", "facility"],
-                "samples": ["name", "description"],
-                "techniques": ["name"],
-            }
+            log.debug("Text operator found within JSON where object")
+            try:
+                search_api_models = importlib.import_module(
+                    "datagateway_api.src.search_api.models",
+                )
+                entity_class = getattr(search_api_models, entity_name)
+            except AttributeError as e:
+                raise SearchAPIError(
+                    f"No text operator fields have been defined for {entity_name}"
+                    f", {e.args}",
+                )
 
             try:
                 or_conditional_filters = []
-                field_names = text_operator_fields[entity_name]
+                field_names = entity_class._text_operator_fields
+                log.debug(
+                    "Text operators found for PaNOSC %s: %s", entity_name, field_names,
+                )
+                if not field_names:
+                    # No text operator fields present, raise KeyError to be caught in
+                    # this try/except block
+                    log.warning(
+                        "No text operator fields found for PaNOSC entity %s, will"
+                        " ignore",
+                        entity_name,
+                    )
+                    raise KeyError()
```

Comment:
Yeah that's a cleaner solution. I've changed to log info instead of a warning (I originally had it as a warning, but this isn't incorrect/worrying behaviour so I think info is sufficient). Changed in d3687781fb880ce716618e9df94288ab2b45c25c